sys/riscv/riscv/exception.S

```diff
@@ -296,7 +325,9 @@ __FBSDID("$FreeBSD$");
 	clc	ca7, (TF_A + 7 * 16)(csp)
 
 	cincoffset csp, csp, (TF_SIZE)
+#ifndef __CHERI_PURE_CAPABILITY__
 .option pop
+#endif
```

Comment:
Nicer to make the push unconditional so this can also be unconditional IMO (and probably repeated elsewhere)

Reply:
I think we have to cleanup this file a bit more as @arichardson commented above https://github.com/CTSRD-CHERI/cheribsd/pull/944#discussion_r584798125 I believe we should make heavier use of macros in exceptions and context switching so that essentially all purecap kernel ifdefs get wrapped. My current take is that I'd prefer to land initial functionality and then refactor this later.

Reply:
I agree in general though it'd be nice to tidy up this minor thing first, which will probably have to stay anyway in some form

Reply:
Should be better now

Reply:
Yep thanks